docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -8,236 +8,342 @@ title: Authorization
 
-- [OAuth 2.1 IETF DRAFT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12)
+- [OAuth 2.1 IETF DRAFT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12)  
 - OAuth 2.0 Authorization Server Metadata
-  ([RFC8414](https://datatracker.ietf.org/doc/html/rfc8414))
+  ([RFC 8414](https://datatracker.ietf.org/doc/html/rfc8414))  
 - OAuth 2.0 Dynamic Client Registration Protocol
-  ([RFC7591](https://datatracker.ietf.org/doc/html/rfc7591))
+  ([RFC 7591](https://datatracker.ietf.org/doc/html/rfc7591))  
+- OAuth 2.0 Demonstrating Proof of Possession (DPoP)
+  ([RFC 9449](https://datatracker.ietf.org/doc/html/rfc9449))  
+- Resource Indicators for OAuth 2.0
+  ([RFC 8707](https://www.rfc-editor.org/rfc/rfc8707.html))
+
+This specification **DOES NOT** require MCP server developers to implement their own Authorization Server
+and instead relies on developers adopting dedicated and well-tested third-party AS implementations.
+
+This specification also enables existing deployments to leverage their current infrastructure.
+As a side-effect, this also simplifies consent flows, and services that have multiple MCP Servers
+won't require individual registration per client:server pair.
 
 ## 2. Authorization Flow
 
 ### 2.1 Overview
 
-1. MCP auth implementations **MUST** implement OAuth 2.1 with appropriate security
-   measures for both confidential and public clients.
+1. Authorization servers **MUST** implement OAuth 2.1 with appropriate security measures
+   (see [RFC 9700](https://datatracker.ietf.org/doc/html/rfc9700)) for both confidential and
+   public clients.
+
+2. Authorization servers **SHOULD** support the OAuth 2.1 Dynamic Client Registration Protocol
+   ([RFC 7591](https://datatracker.ietf.org/doc/html/rfc7591)).
+
+3. Authorization servers **SHOULD** implement and MCP clients **MUST** consume OAuth 2.0 Authorization
+   Server Metadata ([RFC 8414](https://datatracker.ietf.org/doc/html/rfc8414)).
+
+### 2.2 Client OAuth 2.1 Authorization
+
+When authorization is required and not yet proven by the client, servers **MUST** respond with
+*HTTP 401 Unauthorized*.
 
-2. MCP auth implementations **SHOULD** support the OAuth 2.0 Dynamic Client Registration
-   Protocol ([RFC7591](https://datatracker.ietf.org/doc/html/rfc7591)).
+A *HTTP 401 Unauthorized* response **MUST** include a `WWW-Authenticate` header with
+the Authorization Server's issuer URL that the MCP Server is using in the following format:
 
-3. MCP servers **SHOULD** and MCP clients **MUST** implement OAuth 2.0 Authorization
-   Server Metadata ([RFC8414](https://datatracker.ietf.org/doc/html/rfc8414)). Servers
-   that do not support Authorization Server Metadata **MUST** follow the default URI
-   schema.
+`WWW-Authenticate: Bearer realm="default", issuer="https://authserver.example",`  
+`scope="mcp_access_tool1", resource="https://mcp-server.example"`  
```

Comment:
What is the purpose of including the `resource` parameter here? Isn't this always going to be the same hostname as the server this request was made to?

Reply:
Not relevant anymore with move to PRM.